apps/frontend/core-dapp/src/features/trade/TradeStore.ts

```diff
@@ -295,6 +369,16 @@ export class TradeStore {
     })
   }
 
+  async approveClosePositions(): Promise<void> {
+    if (!this.selectedPosition || this.approvingClosePositions) return
+    this.approvingClosePositions = true
+    await this.selectedPosition.token.unlockPermanently('UNISWAP_SWAP_ROUTER')
+    runInAction(() => {
+      this.approvingClosePositions = false
+    })
+  }
+
+  // delete this function when we remove ClosePositionSummary
```

Comment:
nit: use 'TODO' in these comments so they're easy to find later
```suggestion
  // TODO: delete this function when we remove ClosePositionSummary
```